slogr/slogr.go

```diff
@@ -36,14 +36,11 @@ import (
 //
 // The logr verbosity level is mapped to slog levels such that V(0) becomes
 // slog.LevelInfo and V(4) becomes slog.LevelDebug.
+//
+// NewLogr is identical to [logr.NewFromSlogHandler]. Either of these can
+// be used.
 func NewLogr(handler slog.Handler) logr.Logger {
```

Comment:
I don't like that because conceptually, a `var NewLogr` would allow users to replace the function. Not that they'll ever do it, but the API allows it, which feels wrong.

Regarding performance: I'm not sure whether it matters and whether it really makes a difference. It could also backfire because the Go compiler no longer inline the code because `var NewLogr` could be anything.

I'd love to have a `const NewLogr = logr.NewFromSlogHandler`, but there is no such thing in Go.

Reply:
I got curious and benchmarked both approaches. No significant difference.

Reply:
Yeah, I didn't mean benchmark, just one less call frame to debug thru.  You're right that mutability is unfortunate here.  OK.